client/dashboard/sites/settings/danger-zone.tsx

```diff
@@ -48,12 +49,41 @@ const SiteLeaveAction = ( { site }: { site: Site } ) => {
 	);
 };
 
+const showSiteDeleteAction = ( site: Site ) => ! site.is_wpcom_staging_site;
+
+const SiteDeleteAction = ( { site }: { site: Site } ) => {
+	const [ isOpen, setIsOpen ] = useState( false );
+
+	return (
+		<>
+			<ActionList.ActionItem
+				title={ __( 'Delete site' ) }
+				description={ __(
+					"Delete all your posts, pages, media, and data, and give up your site's address."
+				) }
+				actions={
+					<Button
+						variant="secondary"
+						size="compact"
+						isDestructive
+						onClick={ () => setIsOpen( true ) }
+					>
+						{ __( 'Delete' ) }
+					</Button>
+				}
+			/>
+			{ isOpen && <SiteDeleteModal site={ site } onClose={ () => setIsOpen( false ) } /> }
+		</>
+	);
+};
+
 export default function DangerZone( { site }: { site: Site } ) {
 	const canTransferSite = useCanTransferSite( { site } );
 
 	const actions = [
 		canTransferSite && <SiteTransferAction key="transfer-site" site={ site } />,
 		<SiteLeaveAction key="leave-site" site={ site } />,
+		showSiteDeleteAction( site ) && <SiteDeleteAction key="delete-site" site={ site } />,
```

Comment:
Shouldn't it be `canDeleteSite( site )` from `site-delete-modal`; why do we have two different checks 🤔